tests/test_lightcurve_fitting.py

```diff
@@ -1,483 +1,467 @@
-        self.assertEqual(self.params.param2.values,
-                         ['param2', 234.567, 'free', 200, 300])
-        # Test FileNotFoundError
-        self.assertRaises(FileNotFoundError, Parameters, None,
-                          'non-existent.epf')
-
-
-class TestSimulations(unittest.TestCase):
-    """Test the simulations.py module"""
-    def setUp(self):
-        """Setup for the tests"""
-        pass
-
-    def test_simulation(self):
-        """Test the simulations can be made properly"""
-        # Test to pass
-        npts = 1234
-        time, _, _, _ = simulations.simulate_lightcurve('WASP-107b', 0.1,
-                                                        npts=npts, plot=False)
-        self.assertEqual(len(time), npts)
-
-        # Test to fail
-        self.assertRaises(KeyError, simulations.simulate_lightcurve,
-                          'foobar', 0.1)
+
+class testingMetaClass(S5MetaClass):
+    """Apply testing-relevant defaults to S5MetaClass objects.
+
+    Parameters
+    ----------
+    meta : eureka.S5_lightcurve_fitting.s5_meta.S5MetaClass
+        The meta object to prepare.
+    """
+    def __init__(self, folder=None, file=None, eventlabel=None, **kwargs):
+        super().__init__(folder, file, eventlabel, **kwargs)
+        self.spec_hw = 1
+        self.bg_hw = 2
+        self.fit_par = ''
+        self.fit_method = 'lsq'
+        self.run_myfuncs = []
+        self.topdir = './'
+        self.inputdir = ''
+        self.outputdir = ''
+        self.sharedp = False
+        self.whitep = False
+        self.set_defaults()
+
+# Tests for the parameters.py module
+
+
+def test_parameter(capsys):
+    """Test that a Parameter object can be created"""
+    # Create the parameter
+    pname = 'p1'
+    pval = 12.34
+    ptype = 'free'
+    priorpar1 = 10
+    priorpar2 = 15
+    prior = 'U'
+    param = Parameter(pname, pval, ptype, priorpar1, priorpar2, prior)
+
+    # Test bogus input
+    with pytest.raises(TypeError):
+        Parameter(123)
+    with pytest.raises(ValueError):
+        Parameter('foo', 123, 123)
+
+    # Test the attributes
+    assert param.name == pname
+    assert param.value == pval
+    assert param.ptype == ptype
+    assert param.priorpar1 == priorpar1
+    assert param.priorpar2 == priorpar2
+    assert param.prior == prior
+    assert param.values == [pname, pval, ptype, priorpar1, priorpar2, prior]
+
+
+def test_parameters(capsys):
+    """Test that a Parameters object can be created"""
+    params = Parameters()
+    params.param1 = 123.456, 'free'
+    params.param2 = 234.567, 'free', 200, 300
+
+    # Test the auto attribute assignment
+    assert params.param1.values == ['param1', 123.456, 'free']
+    assert params.param2.values == ['param2', 234.567, 'free', 200, 300]
+    # Test FileNotFoundError
+    with pytest.raises(FileNotFoundError):
+        Parameters(None, 'non-existent.epf')
+
+# Tests for the models.py module
+
+
```

Comment:
Remove extra lines

Reply:
The comments delineate a collection of multiple tests for one module, and are not just for one specific test; as a result, I don't want to have zero spaces between the comment and the subsequent test. And flake8 requires two blank lines immediately above each function, so I can't have just one blank line.